src/video/wayland/SDL_waylandwindow.c

```diff
@@ -994,8 +1151,24 @@ void Wayland_SetWindowSize(_THIS, SDL_Window * window)
     SDL_VideoData *data = _this->driverdata;
     SDL_WindowData *wind = window->driverdata;
     struct wl_region *region;
+#ifdef HAVE_LIBDECOR_H
+    struct libdecor_state *state;
+#endif
 
     if (!(window->flags & SDL_WINDOW_RESIZABLE)) {
+#ifdef HAVE_LIBDECOR_H
+        /* TODO: check if this is really needed as this should be handled by
+         * 'Wayland_SetWindowResizable' and LIBDECOR_ACTION_RESIZE
+         */
+        if (data->shell.libdecor) {
+            libdecor_frame_set_min_content_size(wind->shell_surface.libdecor.frame,
+                                                window->w,
+                                                window->h);
+            libdecor_frame_set_max_content_size(wind->shell_surface.libdecor.frame,
+                                                window->w,
+                                                window->h);
```

Comment:
Boy I'm just spamming this thread aren't I...

Commenting this section fixes the shadow bug on Vulkan. It's possible that the content size limits are unintentionally clamping values needed by the plugin to render content outside of our window rectangle? In any case it seems like your TODO checks out, removing this fixes a bug and it doesn't seem to affect resizability because libdecor has a separate toggle for that feature.

Reply:
Even after this, I still see some mismatching between the content and the decoration size when the chosen resolution is higher than the native display resolution. In this case, the window decoration will appear maximised, while the content will be shown at the chosen resolution, partially outside of the window frame (and window geometry).

Reply:
> Boy I'm just spamming this thread aren't I...

I'm pleased this work is getting so much attention, actually.  :)